src/content/post/2024-08-15-kuzu-v-0.5.1.md

```diff
@@ -9,51 +9,62 @@ authors: ["team"]
-// Scan
-MATCH (c:Comment) RETURN MIN(c.ID), MIN(c.creationDate), MIN(c.locationIP), MIN(c.browserUsed), MIN(c.content), MIN(c.length);
-// 2-hop Join
-MATCH (a:Person)-[:knows]->(b:Person)-[:knows]->(c:Person) RETURN MIN(a.birthday), MIN(b.birthday), MIN(c.birthday);
 // Insert
 CREATE NODE TABLE nodeT(id INT64, name STRING, age INT64, net_worth FLOAT, PRIMARY KEY (id));
 // Pass each record's values as parameters as an individual transaction
 CREATE (:nodeT {id: $id, name: $name, age: $age, net_worth: $net_worth})
+// Scan
+MATCH (c:Comment) RETURN MIN(c.ID), MIN(c.creationDate), MIN(c.locationIP), MIN(c.browserUsed), MIN(c.content), MIN(c.length);
+// 2-hop Join
+MATCH (a:Person)-[:knows]->(b:Person)-[:knows]->(c:Person) RETURN MIN(a.birthday), MIN(b.birthday), MIN(c.birthday);
 ```
 
 | mode       |        on-disk (ms)       |     in-mem (ms)   |
 | ---------- | ------------------------- | ----------------- |
 | COPY       |     34586                 | 14798             |
-| Scan       | 5330 (cold) / 1801 (warm) | 1896              |
-| 2-hop Join |  950 (cold) / 899 (warm)  | 904               |  
 | Insert     | 79313                     | 47814             |
+| Scan       | 5330 (cold) / 1801 (warm) | 1896              |
+| 2-hop Join |  950 (cold) / 899 (warm)  | 904               |
 
-From the table, we can see that:
-- the performance of COPY and large scans during cold runs are much improved under the in-memory mode comared to the on-disk one due to zero disk I/Os.
-- for large scans during warms runs, the performance difference between the two modes is neligible (*1896ms vs. 1801ms*), since needed pages are already cached in our buffer manager.
-- for 2-hop joins, the scans on disk are small, thus performance difference is small too.
-- the performance of insertions is greatly improved due to no writes to the WAL, which involves writing and syncing the disk file.
-
-The ideal usage scenario of in-memory mode is when you don't bother persisting your graphs. In-memory mode can further improve the performance for COPY and write heavy workloads.
+The key takeaways are:
+- The performance of COPY and large scans during cold runs are much improved (from 34.5s to 14.7s in this experiment) under the in-memory mode compared to the on-disk mode. 
+This is due to avoiding all disk I/Os that happen that the on-disk mode has to do to persist the data.
+- Similarly, the performance of insertions is greatly improved (from 79.3s to 47.8s in this experiment) due to no writes to the WAL, which involves writing and syncing the disk file.
+- Scans: ``Cold'' scans, i.e., the initial scans that are done when the database starts and the buffer manager is empty also improve significantly (from 5.3s to 1.8s). For warm scans, the performance is empty.   
+- For large scans during warm runs, the performance difference between the two modes is neligible (*1896ms vs. 1801ms*), since needed pages are already cached in our buffer manager.
```

Comment:
```suggestion
- For large scans during warm runs, the performance difference between the two modes is neligible (*1896ms vs. 1801ms*), since the required pages are already cached in the buffer manager.
```